### PR TITLE
Add Cache-Control header

### PR DESCRIPTION
**Pull request by toshinari123**
fixed issue #11

### CODE REVIEW

**Comment by kiootic** [comment]
@toshinari123 
- The indentation for Go uses tab instead of spaces, please run the Go formatter when you edit the code.
- The compiled binary is accidentally committed into the repo.

**Comment by limouren** [comment]
Just passing by, cache control should be configurable?

**Comment by kiootic** [comment]
This is just a reasonable default for Cache-Control header. Customization of headers would be handled separately.

**Comment by limouren** [comment]
icic. It makes sense.

File: internal/handler/site/site_handler.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ func (h *SiteHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 		h.next.ServeHTTP(w, r)
 
+    w.Header().Set("Cache-Control", "public, max-age=31536000, no-cache")
```
Comment (kiootic) [inline internal/handler/site/site_handler.go:52]: This is ineffective; `h.next` is a chain of middleware, ending with `serveFile`, which writes the static file to response. After writing the static file, headers can no longer be set, so this call would be ignored.

File: internal/handler/site/site_handler.go [hunk 2]
```diff
@@ -71,6 +72,7 @@ func (h *SiteHandler) serveFile(w http.ResponseWriter, r *http.Request) {
 	if info.ContentType != "" {
 		w.Header().Set("Content-Type", info.ContentType)
 	}
+    w.Header().Set("Cache-Control", "public, max-age=31536000, no-cache")
```
Comment (kiootic) [inline internal/handler/site/site_handler.go:75]: As seen in the referenced page, the `Cache-Control` header is used with `ETag` header, so it makes more sense to set it together with the `ETag` header.

**Pull request merged**